File: 백준/Silver/10773. 제로/제로.py
import sys

result = []

N = int(sys.stdin.readline().strip())

for i in range(N):
  temp = int(sys.stdin.readline().strip())
  if temp == 0:
    result.pop(-1)
  else:
    result.append(temp)
    
sum = 0
for i in range(len(result)):
  sum+=result[i]
  
print(sum)

File: 백준/Silver/10773. 제로/제로.java

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;


public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
        String s = bf.readLine();
        int K = Integer.parseInt(s);

        Stack<Integer> stack = new Stack();

        for(int k=0; k<K; k++){
            s = bf.readLine();
            int N = Integer.parseInt(s);

            if (N != 0){
                stack.push(N);
            }else{
                if (stack.size() != 0){
                    stack.pop();
                }
            }
        }
        
        int result = 0;
        while (stack.size() != 0){
            result += stack.pop();
        }

        System.out.println(result);
    }
}
